scripts/editor/register-blocks.js

```diff
@@ -483,7 +483,14 @@ const registerVariation = (
 ) => {
 
 	// Append globalManifest data in to output.
-	blockManifest['icon'] = getIconOptions(globalManifest, blockManifest);
+	if (blockManifest['icon']) {
+		blockManifest['icon'] = getIconOptions(globalManifest, blockManifest);
+	} else {
+		// There is no icon passed to variation, use it's parent icon instead
```

Comment:
```suggestion
		// If no icon is passed to the block variation, use the parent block icon instead.
```

Reply:
the PR approach is wrong so I will remove this comment because it needs to be refactored.